.github/workflows/check.yaml

```diff
@@ -31,7 +31,7 @@ jobs:
     - name: Linters
       uses: golangci/golangci-lint-action@v3
       with:
-        version: v1.48.0
+        version: v1.54.2
```

Comment:
In light of the new complaints by the linter, I think the simplest way forward is to reset this back, and focus on the original 3 reported issues.

Reply:
I think depguard could be disabled or otherwise I could add configuration for it. The errors are due to the new version: https://github.com/golangci/golangci-lint/issues/3877#issuecomment-1573682848

Reply:
I'm OK with either if you are.